### PR TITLE
Fulltext index 

### PR DESCRIPTION
**Pull request by cpegeric**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # #18869

## What this PR does / why we need it:

Implement fulltext index search for v2.

The design of the fulltext index search
https://github.com/matrixorigin/docs/blob/main/design/mo/sql/20240912-cpegeric-fulltext.md

___

### **PR Type**
Enhancement, Feature, Tests, Bug fix, Dependencies, Configuration changes

___

### **Description**
- Implemented full-text index search functionality, including index scan and tokenization.
- Enhanced query planning and execution to integrate full-text search capabilities.
- Added new built-in functions `FULLTEXT_MATCH` and `FULLTEXT_MATCH_SCORE` for full-text matching.
- Enhanced DDL operations to support full-text indices, including creation and validation.
- Updated index utilities and SQL generation logic to accommodate full-text indices.
- Added extensive test cases for full-text search pattern matching and syntax.
- Introduced a system variable `experimental_fulltext_index` for managing experimental features.
- Fixed a directory check in the proto generation script.
- Updated module dependencies to support full-text search implementation.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>fulltext_test.go</strong><dd><code>Add tests for full-text search pattern matching and evaluation</code></dd></summary>
<hr>

pkg/sql/colexec/table_function/fulltext_test.go

<li>Added a new test file for full-text search functionality.<br> <li> Defined multiple test cases for different pattern matching scenarios.<br> <li> Implemented functions to test pattern conversion and evaluation.<br> <li> Included tests for various full-text search modes and operators.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-5dada9ee500589e4def52e38a1fd55fac5295b9da25bfc6ad4be56e399394866">+741/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_sql_test.go</strong><dd><code>Add test cases for full-text search syntax</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/dialect/mysql/mysql_sql_test.go

- Added test cases for full-text search queries.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-96a7d142d5e6047d7d72e9a45bc7e4a8baeaa0be0ddfece9753e336a09d43ec2">+24/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fulltext.result</strong><dd><code>Add test results for full-text search cases</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/distributed/cases/fulltext/fulltext.result

- Added test results for full-text search functionality.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-2045017efbb026c4485d17249d8cc8ad338eeaad5280eeb33f33a8dc8a3d4bd8">+103/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fulltext.sql</strong><dd><code>Add SQL test cases for full-text search functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/distributed/cases/fulltext/fulltext.sql

- Added SQL test cases for full-text search functionality.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-6b100c429a12e9b9fb9c28f9617e3530df027ea45c4701dc0c20948f4f39b3be">+81/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>18 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>fulltext_util.go</strong><dd><code>Implement utility functions for full-text search operations</code></dd></summary>
<hr>

pkg/sql/colexec/table_function/fulltext_util.go

<li>Implemented utility functions for full-text search pattern parsing and <br>evaluation.<br> <li> Defined data structures for managing search patterns and results.<br> <li> Added functions to handle different full-text search operators.<br> <li> Included error handling for invalid patterns and operations.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-84ed1366fcf1e3986b78bf82fa01931651fdf7df592366059f0ebc25375f6267">+735/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>apply_indices_fulltext.go</strong><dd><code>Integrate full-text index application in query planning</code>&nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/apply_indices_fulltext.go

<li>Added logic to apply full-text indices in query plans.<br> <li> Implemented functions to handle full-text match and index scan.<br> <li> Integrated full-text search into query planning and execution.<br> <li> Provided support for sorting and joining based on full-text scores.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-089afff5c61c7f77e1346b6f308ae2a8625c1e346ff0873698871c6d3649dd0d">+581/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build_ddl.go</strong><dd><code>Enhance DDL building to support full-text indices</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/build_ddl.go

<li>Enhanced DDL building to support full-text indices.<br> <li> Added logic to handle full-text index creation and validation.<br> <li> Implemented support for full-text index options and parameters.<br> <li> Modified index validation to accommodate full-text indices.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-cf52fe4dff6553ddcdc78a208d8fb66504d93633b1d9eb23b92068a1c4d56357">+231/-5</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fulltext.go</strong><dd><code>Implement full-text index scan table function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/colexec/table_function/fulltext.go

<li>Implemented full-text index scan table function.<br> <li> Added logic to execute full-text search and scoring.<br> <li> Defined structures and methods for handling search results.<br> <li> Integrated with SQL execution process for full-text queries.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-1130c1ca01e40c936a3a2865d37a6bed7e94247e8cf390ba0fb71b5ca2747a24">+289/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>secondary_index_utils.go</strong><dd><code>Update index utilities to support full-text indices</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/catalog/secondary_index_utils.go

<li>Updated index utility functions to support full-text indices.<br> <li> Added logic to handle full-text index parameters and options.<br> <li> Implemented checks for full-text index algorithm support.<br> <li> Refactored index parameter handling for flexibility.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-02b5c26b2b519abc61917055819fc19c076586fd4616ab0cc23b13dc00c32d54">+68/-39</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>util.go</strong><dd><code>Add SQL generation and insertion logic for full-text indices</code></dd></summary>
<hr>

pkg/sql/compile/util.go

<li>Added SQL generation for full-text index table creation.<br> <li> Implemented logic to insert data into full-text index tables.<br> <li> Enhanced utility functions to support full-text index operations.<br> <li> Included SQL formatting for full-text index management.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-a25090ac97106b09200abf43241be92bc716b7a47f5185748e012fa8a03f1336">+81/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>apply_indices_vector.go</strong><dd><code>Enhance index resolution and application in query plans</code>&nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/apply_indices_vector.go

<li>Modified logic to resolve scan nodes with indices.<br> <li> Added support for resolving scan nodes from project nodes.<br> <li> Enhanced index application logic for vector-based queries.<br> <li> Improved handling of sort and aggregation nodes in plans.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-645377b8dc2147e6a0f3ce6c03f303cb09971a07c6d0cef9d6e6c622ca8c4508">+16/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>apply_indices.go</strong><dd><code>Add full-text index application logic in query builder</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/apply_indices.go

<li>Added logic to apply indices for full-text search in <br><code>applyIndicesForProject</code>.<br> <li> Enhanced <code>applyIndices</code> function with additional checks and logic.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-b92e434ebc0c08092943a8c5a8abaed315f3e5722db3ffde4e3fbbb878d325b2">+65/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>vector.go</strong><dd><code>Add generic GetAny function for vector type retrieval</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/container/vector/vector.go

- Added `GetAny` function to retrieve any type from a vector.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-28ac70822524921592389604c9d3caa2e7c488a52b02d22fe9bdc35f8d808d5f">+55/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ddl.go</strong><dd><code>Add full-text index handling in DDL operations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/compile/ddl.go

<li>Added handling for full-text index in <code>AlterTableInplace</code> and <br><code>CreateIndex</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-819bfdf891fa506cb2b4c89006896d528675fd554c341f38d1a1bf14a496c8aa">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>base_binder.go</strong><dd><code>Implement binding for full-text match expressions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/base_binder.go

- Added binding logic for `FullTextMatchExpr`.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-d2c4ebed110cf4c8982fa501ff026b2d518810445a72f51eac3021b51396e7b0">+20/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>query_builder.go</strong><dd><code>Extend query builder for full-text index functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/query_builder.go

<li>Added cases for <code>fulltext_index_scan</code> and <code>fulltext_index_tokenize</code> in <br><code>buildTableFunction</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-144e02a38da50867dc021b9254d10a5fd131671cca344c2323337ba2141440f1">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>types.go</strong><dd><code>Define constants for full-text index columns</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/catalog/types.go

- Added constants for full-text index table columns.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-1695ed5765b08f4366c290109026c6d6d3ce15b7a0fc0a5ad86b6646c92e81a7">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>table_function.go</strong><dd><code>Integrate full-text index operations in table functions</code>&nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/colexec/table_function/table_function.go

<li>Added constants for full-text index scan and tokenize operations.<br> <li> Integrated full-text index preparation logic.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-996a7154a7ffac18323cc9172fee85b20b1dcd278af5cc27e1a0d2a56f971cf1">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build_index_util.go</strong><dd><code>Implement naming utility for full-text indices</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/build_index_util.go

<li>Added <code>setEmptyFullTextIndexName</code> function for naming full-text indices.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-dab55661e65fc12d002528146a96e1d4894651650ccc3249eb14d3a4b6b3462b">+20/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>create.go</strong><dd><code>Add full-text index type to index definitions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/tree/create.go

- Added `INDEX_TYPE_FULLTEXT` to index types.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-56c8b9be8afe4a5d132d5f3bcb9f4bb82164492bb538b173bb0181eafcec199e">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mysql_sql.y</strong><dd><code>Extend SQL grammar for full-text search options</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/dialect/mysql/mysql_sql.y

- Added grammar rules for full-text search options.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-3e21a81f1ee10d34359f5ed2d3a4cf12a1b0aacaa6885b141b4c1022310345b7">+41/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>plan.proto</strong><dd><code>Extend IndexOption with full-text search parameters</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

proto/plan.proto

<li>Added fields for parser name and ngram token size in <code>IndexOption</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-22bd2bd1e47d7acf42499abc93266a657dfd1a35f32837167470fd926564dfb5">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Feature</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>fulltext.go</strong><dd><code>Implement full-text index scan and tokenize functions</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/fulltext.go

<li>Introduced new functions <code>buildFullTextIndexScan</code> and <br><code>buildFullTextIndexTokenize</code> for full-text index operations.<br> <li> Defined column definitions for full-text index and tokenization.<br> <li> Added a helper function <code>getFullTextParams</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-455d3c963d4ba5a0afa98d5cd338324f078c8a1a0ae2e62e5dccc1a5a97d93ed">+155/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fulltext_tokenize.go</strong><dd><code>Implement full-text tokenization logic for table functions</code></dd></summary>
<hr>

pkg/sql/colexec/table_function/fulltext_tokenize.go

<li>Implemented <code>tokenizeState</code> struct and related methods for full-text <br>tokenization.<br> <li> Added <code>fulltextIndexTokenizePrepare</code> function for preparing <br>tokenization.<br> <li> Introduced <code>start</code> method for tokenizing document content.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-5bc64741f1a51fb3405f42631f0ab31edcd6af78a6d518cce6c86ce27ee4c33d">+135/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>list_builtIn.go</strong><dd><code>Add built-in functions for full-text matching</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/function/list_builtIn.go

<li>Added new built-in functions <code>FULLTEXT_MATCH</code> and <code>FULLTEXT_MATCH_SCORE</code>.<br> <li> Expanded function overloads for full-text matching.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-b9b859604bf3c01505320f0a72c9862667f39cdf1323a88d7f8b2991d32098b8">+101/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>expr.go</strong><dd><code>Define full-text search expression and types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/tree/expr.go

<li>Introduced <code>FullTextMatchExpr</code> struct for full-text search expressions.<br> <li> Added <code>FullTextSearchType</code> constants for different search modes.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-6f8d2e3805b41b9704c5d638c08078e8a7e5f6812ffa344ad607746319026e1f">+78/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ddl_index_algo.go</strong><dd><code>Implement full-text index table handling logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/compile/ddl_index_algo.go

<li>Implemented <code>handleFullTextIndexTable</code> for managing full-text index <br>tables.<br> <li> Added experimental flag check for full-text index.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-c6f4b272afb302e3b5d0d6460544032c03ac0ec988fd3cc8e89389ba657c0477">+30/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>func_fulltext.go</strong><dd><code>Add placeholder functions for full-text match operations</code>&nbsp; </dd></summary>
<hr>

pkg/sql/plan/function/func_fulltext.go

<li>Added placeholder functions for <code>fullTextMatch</code> and <code>fullTextMatchScore</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-e615293b5a5b2ee465b7b024c8ba8aea33f45c5e58d726ae786b97554b2cb06b">+29/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>function_id.go</strong><dd><code>Define function IDs for full-text match functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/plan/function/function_id.go

<li>Added function IDs for <code>FULLTEXT_MATCH</code> and <code>FULLTEXT_MATCH_SCORE</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-8a340cc7bfd15d211dc0f230d234ba4edded44e79655671a94ce13f5fabfc7de">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>variables.go</strong><dd><code>Introduce system variable for experimental full-text index</code></dd></summary>
<hr>

pkg/frontend/variables.go

- Added system variable `experimental_fulltext_index`.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-07649adc637b0268b12ec915ec819f66db3a4e4375bfc77291116bfa06c4432f">+9/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>gen.sh</strong><dd><code>Fix directory check in proto generation script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

proto/gen.sh

- Corrected directory check in script.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-c7f9d9ef33c478fb9571eac41f35f98f3615dc0aaa9076c7d0fbb85b8f7d57ab">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>go.sum</strong><dd><code>Update dependencies for full-text search implementation</code>&nbsp; &nbsp; </dd></summary>
<hr>

go.sum

- Updated dependencies for `monlp` and other modules.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-3295df7234525439d778f1b282d146a4f1ff6b415248aaac074e8042d9f42d63">+10/-8</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>go.mod</strong><dd><code>Update module dependencies for full-text search support</code>&nbsp; &nbsp; </dd></summary>
<hr>

go.mod

- Updated module dependencies for full-text search.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6">+5/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Additional files (token-limit)</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>mysql_sql.go</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/dialect/mysql/mysql_sql.go

...

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/18870/files#diff-6aa8b3d2cec1bd35bafe74d27392cee02153f7b4ec574d40eff17db7c30d41e2">+6034/-5947</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

**Review by m-schen — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Review by badboynt1 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/container/vector/vector.go [hunk 1]
```diff
@@ -342,6 +342,61 @@ func GetArrayAt[T types.RealNumbers](v *Vector, i int) []T {
 	return types.GetArray[T](&bs[i], v.area)
 }
 
+func GetAny(vec *Vector, i int) any {
```
Comment (XuPeng-SH) [inline pkg/container/vector/vector.go:345]: It is not recommended to add this GetAny interface, the use of `any` will have additional on-heap memory allocation, causing stress on the GC. Similarly the use of AppendAny is not recommended.

File: pkg/sql/colexec/table_function/fulltext_tokenize.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec"
+	"github.com/matrixorigin/matrixone/pkg/vm"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"github.com/matrixorigin/monlp/tokenizer"
+)
+
+type FullTextEntry struct {
+	DocId any
```
Comment (XuPeng-SH) [inline pkg/sql/colexec/table_function/fulltext_tokenize.go:34]: Replay with
```golang
type FullTextEntry struct {
     DocId []byte
     Type types.T
     .....
}
```
`GetAny` and `AppendAny` are very convenient, but have side effects, it is recommended to use them only in some TEST scenarios, it is best to delete these two

**Review by XuPeng-SH — CHANGES_REQUESTED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

File: pkg/container/vector/vector.go (hunk 1 above)
Comment (cpegeric) [reply pkg/container/vector/vector.go:348]: comment added for GetAny and AppendAny to warn that memory will escape to heap for any type that result in slow GC.

File: pkg/sql/colexec/table_function/fulltext_tokenize.go (hunk 2 above)
Comment (cpegeric) [reply pkg/sql/colexec/table_function/fulltext_tokenize.go:34]: comment added for GetAny and AppendAny to warn that memory will escape to heap for any type that result in slow GC.

**Pull request merged**